doc/decisions/0_drafts/errors_core.md

```diff
@@ -0,0 +1,403 @@
+# Errors without `errorKey`
+
+## Problem
+
+We often use a `Key * errorKey` argument to return error information via metadata.
+In `libelektra-core`, however, that would be inappropriate and sometimes even impossible (e.g., `keyNew`).
+
+Not using a `Key * errorKey` argument can be limiting, however.
+Depending on the solution, there can be limited error values, i.e., it may not be possible to use a different error value for every possible error case.
+A prime example of this, a function that returns a pointer and wants to indicate an error via that pointer, only has to option to return `NULL`.
+So if there is more than one possible error, the error cases have to be distinguished another way.
```

Comment:
The question is if this is actually the case, and in which functions. This would be a decision specific enough. But I already said something along the lines below.

Reply:
For example `ksLookup`. The return value `NULL` could mean either "key not found" or "invalid arguments". Technically it may not apply 100%, since I wouldn't consider "key not found" an error case. However, conflating a non-error return value with error cases is arguably even worse.

If we go with the more general case (not just limited to pointer return value), then there is also `ksSearch`. Where `-1` means basically the same as `NULL` for `ksLookup`.

If you want a very clear-cut example in the current API then just look at `keySetName`. It definitely conflates multiple error cases into `-1` (read-only key, invalid name, etc.).